src/main/java/dao/impl/BasketDaoJdbcImpl.java

```diff
@@ -0,0 +1,103 @@
+package dao.impl;
+
+import dao.interfaces.BasketDao;
+import model.Product;
+import model.User;
+import org.apache.log4j.Logger;
+import utils.DBConnection;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BasketDaoJdbcImpl implements BasketDao {
+
+    private static final Logger logger = Logger.getLogger(BasketDaoJdbcImpl.class);
+
+    @Override
+    public void addProduct(Product product, User user) {
+        String sql = String.format(
+                "INSERT INTO basket(product_id, product_name, " +
```

Comment:
now better use constants for sql queries

Reply:
I fixed it.